src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -231,6 +231,77 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::CreateNew
     return *newProfile;
 }
 
+// Method Description:
+// - Duplicate a new profile based off another profile's settings
+// Arguments:
+// - source: the Profile object we are duplicating (must not be null)
+// Return Value:
+// - a reference to the new profile
+winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::DuplicateProfile(Model::Profile source)
+{
+    THROW_HR_IF_NULL(E_INVALIDARG, source);
+
+    auto duplicated{ _userDefaultProfileSettings->CreateChild() };
+    _allProfiles.Append(*duplicated);
+
+    if (!source.Hidden())
+    {
+        _activeProfiles.Append(*duplicated);
+    }
+
+    const auto newName{ fmt::format(L"{} ({})", source.Name(), RS_(L"CopySuffix")) };
+    duplicated->Name(winrt::hstring(newName));

```

Comment:
```suggestion
    const winrt::hstring newName{ fmt::format(L"{} ({})", source.Name(), RS_(L"CopySuffix")) };
    duplicated->Name(newName);
```

this saves you a needless cast.